tests/utils.py

```diff
@@ -820,8 +822,30 @@ class SnapshotFailed(Exception):
     pass
 
 
+@attr.s
+class SnapshotTest(object):
+    token = attr.ib(type=str)
+    tracer = attr.ib(type=ddtrace.Tracer, default=ddtrace.tracer)
+
+    def clear(self):
+        """Clear any traces sent that were sent for this snapshot."""
+        parsed = parse.urlparse(self.tracer.writer.agent_url)
+        conn = httplib.HTTPConnection(parsed.hostname, parsed.port)
+        conn.request("GET", "/test/clear?token=%s" % self.token)
+        resp = conn.getresponse()
+        assert resp.status == 200
+
+
 @contextmanager
-def snapshot_context(token, ignores=None, tracer=None, async_mode=True):
+def snapshot_context(token, ignores=None, tracer=None, async_mode=True, variants=None):
+    # Use variant that applies to update test token. One must apply. If none
+    # apply, the test should have been marked as skipped.
+    if variants:
+        applicable_variant_ids = [k for (k, v) in variants.items() if v]
+        assert len(applicable_variant_ids) == 1
+        variant_id = applicable_variant_ids[0]
+        token = "{}_{}".format(token, variant_id) if variant_id else token
```

Comment:
I realize this has already been in the code but I wonder how this interacts with the token for a test that has parameters. E.g., `test_foo[p0-p1]`.

Reply:
Yeah it doesn't at all - it seems like the name comes from pytest. pytest doesn't do anything too crazy like modifying the actual function names or anything.